lib/puppet/indirector/request.rb

```diff
@@ -250,8 +250,6 @@ def set_uri_key(key)
       return
     end
 
-    env, indirector, @key = URI.unescape(uri.path.sub(/^\//, '')).split('/',3)
-    @key ||= ''
-    self.environment = env unless env == ''
+    @key = key
```

Comment:
Ah, this is actually not dead code. When using a remote file bucket, the uri is of the form:

```
(rdb:2) uri.to_s
"https://hwvft1j1a4531gm.delivery.puppetlabs.net:8140/production/file_bucket_file/md5/d41d8cd98f00b204e9800998ecf8427e//Users/josh/work/puppet/empty"
```

but due to the code above, the request key is rewritten, and later the indirected routes transforms the HTTP request to:

```
$ bundle exec puppet filebucket backup empty --server hwvft1j1a4531gm.delivery.puppetlabs.net --http_debug
...
<- "HEAD /puppet/v3/file_bucket_file/md5/d41d8cd98f00b204e9800998ecf8427e//Users/josh/work/puppet/empty?environment=production& HTTP/1.1
...
```

So this raises some interesting questions:

1. We haven't specified what a valid file_bucket request/response look like in api/schemas. We should do that before making any changes.
2. The Dipper gets the environment (as a string) from `Puppet[:environment]`, includes it in the URI prefix, and the Request strips it out. That is confusing and unnecessary. If an environment is specified in defaults.rb, as a setting, or command-line, then puppet will make that the `Puppet.lookup(:current_environment)`. And since the request will use that environment, there is no reason for the Dipper to specify it when calling the indirected methods, e.g. `Puppet::FileBucket::File.indirection.head(file_bucket_path)`. IOW, the Dipper shouldn't need to specify the enviroment.